ddtrace/settings/config.py

```diff
@@ -552,7 +552,6 @@ def __init__(self):
         self._telemetry_install_time = os.getenv("DD_INSTRUMENTATION_INSTALL_TIME", None)
 
         self._dd_api_key = os.getenv("DD_API_KEY")
-        self._dd_app_key = os.getenv("DD_APP_KEY")
```

Comment:
This is a significant enough change to the public interface that I think it deserves a deprecation warning.

Reply:
@emmettbutler While I would agree, there's some context here that I think makes this actually not so significant and justifies no release notes:
- `DD_APP_KEY` was a private config option for the openai/bedrock integrations for LLMObs users to submit data to LLMObs. This was later refactored to being a global config object in #8254, but since this was only for LLMObs users (currently private beta phase) we didn't add any release notes to add it as part of our public interface.
- With the recent change to LLMObs intake, we actually don't need `DD_APP_KEY` anymore to submit LLMObs payloads. This means this env var is unnecessary (unfortunately only found this out after #8254 was merged 😢)
- LLMObs is still in a private design partnership phase right now, meaning we don't need to worry about deprecating this change since there aren't any active users submitting LLMObs data.

The takeaway for me here is to be more careful adding to our public interface in the future, but in the meantime I don't see any reason that we shouldn't delete this config entry which wasn't public to begin with. WDYT?

Reply:
Makes sense to me - no deprecation necessary. Thanks for explaining.